example/particles/particles.cpp

```diff
@@ -139,11 +143,25 @@ TaskStatus DestroySomeParticles(MeshBlock *pmb) {
   return TaskStatus::complete;
 }
 
+TaskStatus SortParticlesIfNecessary(MeshBlock *pmb) {
+  auto pkg = pmb->packages.Get("particles_package");
+  const auto deposition_method = pkg->Param<std::string>("deposition_method");
+  if (deposition_method == "per_cell") {
```

Comment:
Is this the "Necessary" condition?

Reply:
Yeah, since these sorted arrays are used by `"per_cell"` deposition in the next task